codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -133,8 +140,15 @@ public int initialWindowSize(Http2Stream stream) {
         return state(stream).initialWindowSize();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This method must be called from the same event loop context as the {@link ChannelHandlerContext}
```

Comment:
@nmittler - At one point there was interest in providing a "thread safe" wrapper around `Http2FrameWriter`'s `writeXXX` methods.  The idea would be to mimic the thread safety provided by Netty's `channel.write(..)` methods.  I have not gone down this path yet but I'm guessing this would most likely result in a some sort of decorator around `DefaultHttp2ConnectionEncoder` (and maybe useful to provide something similar for `DefaultHttp2FrameWriter` too).

So this PR does not attempt to completely document all the interfaces in the HTTP/2 codec, but instead focuses on the flow controllers.